src/string.cr

```diff
@@ -913,7 +913,7 @@ class String
   def downcase(options = Unicode::CaseOptions::None)
     return self if empty?
 
-    if ascii_only? && (options.none? || options.ascii?)
+    if ascii_only?
```

Comment:
Are you sure about this one?

Reply:
I removed this from `String#upcase`, `String#capitalize` and from `String#downcase` and all specs failed except for this method. And from all three this one actually has the most specs: [`spec/std/string_spec.cr:476`](https://github.com/crystal-lang/crystal/blob/master/spec/std/string_spec.cr#L476). It especially has many that use the `options` argument. But I could maybe add some more specs to `String#downcase` to be 100 percently sure.

Reply:
I prefer to be sure because shaving 2 integer checks after `ascii_only?` that iterated the whole string should have a negligible impact on performance.

Reply:
Alright there's no difference anyway.
```
    downcase 300.45  (  3.33ms) (±14.33%)  350256 B/op        fastest
new_downcase 286.03  (   3.5ms) (±15.35%)  350257 B/op   1.05× slower
```